tests/conftest.py

```diff
@@ -1,37 +1,92 @@
 import json
 from datetime import datetime, timezone
+from unittest.mock import MagicMock, patch
 
+import boto3
 import pytest
+from botocore.stub import Stubber
+
+# Usage and cost data fixtures for test_cost.py
 
 
 @pytest.fixture(scope="function")
 def input_data_usage():
-    with open("tests/test_data_usage.json") as f:
+    with open("tests/data/test_data_usage.json") as f:
         data = json.load(f)
     return data
 
 
 @pytest.fixture(scope="function")
 def input_data_cost():
-    with open("tests/test_data_cost.json") as f:
+    with open("tests/data/test_data_cost.json") as f:
         data = json.load(f)
     return data
 
 
 @pytest.fixture(scope="function")
 def output_data_hub():
-    with open("tests/test_output_hub.json") as f:
+    with open("tests/data/test_output_hub.json") as f:
         data = json.load(f)
     return data
 
 
 @pytest.fixture(scope="function")
 def output_data_component():
-    with open("tests/test_output_component.json") as f:
+    with open("tests/data/test_output_component.json") as f:
         data = json.load(f)
     return data
 
 
+# Usage and cost data for test_integration.py
+
+
+@pytest.fixture(scope="function", params=["compute", "home_storage"])
+def mock_usage_response(request):
+    """
+    Mock Prometheus response for compute and home storage components.
+    """
+    with patch("requests.get") as mock_get:
+        mock_response = MagicMock()
+        with open(f"tests/data/test_data_usage_{request.param}.json") as f:
+            mock_response.json.return_value = json.load(f)
+        mock_get.return_value = mock_response
+        mock_get.test_param = request.param
+        yield mock_get
+
+
+@pytest.fixture(scope="function")
+def mock_ce():
+    """
+    Mock multiple responses from the AWS Cost Explorer client to validate cost logic of `query_total_costs_per_user` function in `query_cost_aws` submodule.
+
+    Query parameters used to generate json test data files:
+
+    ```
+    {
+        "TimePeriod": {"Start": "2025-09-01", "End": "2025-09-03"},
+        "Granularity": "DAILY",
+        "Metrics": ["UnblendedCost"],
+        "GroupBy": [{"Type": "DIMENSION", "Key": "SERVICE"}],
+        "Filter": base_filter/home_storage_filter/core_filter,
+    }
+    ```
+    """
+    aws_ce_client = boto3.client(
+        "ce",
+        region_name="us-west-2",
+        aws_access_key_id="fake-key",
+        aws_secret_access_key="fake-secret",
+        aws_session_token="fake-token",
+    )  # Use fake credentials, otherwise boto3 will use real ones from your environment
+    stubber = Stubber(aws_ce_client)
+    for c in ["all", "home_storage", "core"]:
+        with open(f"tests/data/test_data_cost_component_{c}.json") as f:
+            response = json.load(f)
+        stubber.add_response("get_cost_and_usage", response)
+    with stubber:
+        yield aws_ce_client
```

Comment:
if we patch the `aws_ce_client` from `jupyterhub_cost_monitoring.query_cost_aws` directly, we won't need to pass the mock client to the individual functions as an argument:
```suggestion
        with patch("src.jupyterhub_cost_monitoring.query_cost_aws.aws_ce_client", aws_ce_client):
            yield aws_ce_client
```